e2e/publish_test.go

```diff
@@ -28,22 +28,22 @@ type cleanup struct {
 	volumes map[string]string
 }
 
-func (c cleanup) register(volumeID, targetPath string) {
+func (c *cleanup) register(volumeID, targetPath string) {
```

Comment:
This caused the CI to fail.

https://github.com/topolvm/topolvm/runs/2719285981

```
Run make test
test -z "$(gofmt -s -l . | grep -v '^vendor' | tee /dev/stderr)"
/home/runner/work/topolvm/topolvm/bin/staticcheck ./...
e2e/publish_test.go:58:2: ineffective assignment to field cleanup.volumes (SA4005)
Makefile:101: recipe for target 'lint' failed
make: *** [lint] Error 1
Error: Process completed with exit code 2.
```

Reply:
https://staticcheck.io/docs/checks#SA4005

Reply:
If the point is incorrect, we can disable it according to the documentation below.
https://staticcheck.io/docs/configuration#maintenance-of-linter-directives

Reply:
This case is a code bug so I think it's correct to fix it.
The map update I'm doing here is meaningless because the receiver isn't a pointer.
https://github.com/topolvm/topolvm/blob/v0.8.3/e2e/publish_test.go#L31-L37